scripts/code_style.py

```diff
@@ -137,6 +139,20 @@ def get_src_files(since: Optional[str]) -> List[str]:
                          is_file_autogenerated(filename))]
     return src_files
 
+def get_submodule_hash(commit: str, submodule: str) -> str:
+    cmd = ["git", "ls-tree", commit, submodule]
+    output = subprocess.check_output(cmd, universal_newlines=True)
+
+    # Need to extract submodule commit hash from the command output.
+    # Form is: <> commit <hash> <submodule>
+    hash_regex = r"\b[0-9a-f]{40}\b"
+    results = re.search(hash_regex, output)
```

Comment:
Minor: this is a slightly fragile way of extracting the sha. It could fail if we add a submodule whose name just happens to be 40 hex digits, or if git ever starts to emit longer shas. The format of `git ls-tree` defines the sha as the third whitespace-separated field, so we can just do
```
return output.split()[2]
```